### PR TITLE
[SWATANG-6] - implement reset password feature

### LINKED ISSUE #6: Implement Reset password feature

### PR DESCRIPTION
**Pull request by quangloc1993**
### Description
- Implement new feature to reset user password
- Implement unit test for feature component

### ☑️ Definition of Done Checklist
- [x] Keep the application can be launched successfully
- [x] Keep the source code is built successfully
- [x] Keep the unit test can run successfully

### Check the box when resolved these items and provide more information

- [ ] Documentation
- [ ] Localization
- [x] Coding
- [x] Unit Tests
- [ ] Integration Tests
- [ ] System Tests

### 📷 Screenshot / GIF

image
image

### CODE REVIEW

File: src/app/shared/validation/field-validator.ts [hunk 1]
```diff
@@ -36,11 +36,12 @@ export class FieldValidators {
 
   static matchPassword(): ValidatorFn {
     return (control: AbstractControl): { [key: string]: any } | null => {
-      const password = control.get('newpassword').value
-      const confirmPassword = control.get('confirmpassword').value
+      console.log(control);
```
Comment (mwd-luongmai) [inline src/app/shared/validation/field-validator.ts]: remove log

File: src/app/shared/validation/field-validator.ts [hunk 2]
```diff
@@ -36,11 +36,12 @@ export class FieldValidators {
...
+      const password = control.get('password').value
+      const confirmPassword = control.get('confirmPassword').value
       const isValid = password === confirmPassword
       if (!isValid) {
-        control.get('confirmpassword').setErrors({ matchPassword: true })
+        control.get('confirmPassword').setErrors({ matchPassword: true })
```
Comment (mwd-luongmai) [inline src/app/shared/validation/field-validator.ts:43]: this validator is updated. however, don't see the place which is using it????

**Review by mwd-luongmai — APPROVED** [review]
LGTM

**Review by thienkieu — APPROVED** [review]
LGTM

**Pull request merged**